README.md
# custom_card_design

A new Flutter package for design card.

## Getting Started

Use following in ```pubspec.yaml``` for installing this package from git

```
dependencies:
  flutter:
    sdk: flutter
  
  custom_design:
    git:
      url: https://github.com/ezatechbd/flutter_package_card_design.git
      <!-- ref: d3bbf9b78530c3b8959850dee4f54a0d51a7edf6 # commit hash -->
```


CHANGELOG.md
## [0.0.1] - 2022-01-25

* For those who will use our package here
 We state the changes we made by date.
 
* A description for people who will use that package or version.

example/lib/main.dart
import 'package:custom_design/custom_design.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        // This is the theme of your application.
        //
        // Try running your application with "flutter run". You'll see the
        // application has a blue toolbar. Then, without quitting the app, try
        // changing the primarySwatch below to Colors.green and then invoke
        // "hot reload" (press "r" in the console where you ran "flutter run",
        // or simply save your changes to "hot reload" in a Flutter IDE).
        // Notice that the counter didn't reset back to zero; the application
        // is not restarted.
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  // This class is the configuration for the state. It holds the values (in this
  // case the title) provided by the parent (in this case the App widget) and
  // used by the build method of the State. Fields in a Widget subclass are
  // always marked "final".

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() async {
    CustomAlertBox.showCustomAlertBox(context: context, willDisplayWidget: Text('Show dialog'));
    setState(() {
      // This call to setState tells the Flutter framework that something has
      // changed in this State, which causes it to rerun the build method below
      // so that the display can reflect the updated values. If we changed
      // _counter without calling setState(), then the build method would not be
      // called again, and so nothing would appear to happen.
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    // This method is rerun every time setState is called, for instance as done
    // by the _incrementCounter method above.
    //
    // The Flutter framework has been optimized to make rerunning build methods
    // fast, so that you can just rebuild anything that needs updating rather
    // than having to individually change instances of widgets.
    return Scaffold(
      appBar: AppBar(
        // Here we take the value from the MyHomePage object that was created by
        // the App.build method, and use it to set our appbar title.
        title: Text(widget.title),
      ),
      body: Center(
        // Center is a layout widget. It takes a single child and positions it
        // in the middle of the parent.
        child: Column(
          // Column is also a layout widget. It takes a list of children and
          // arranges them vertically. By default, it sizes itself to fit its
          // children horizontally, and tries to be as tall as its parent.
          //
          // Invoke "debug painting" (press "p" in the console, choose the
          // "Toggle Debug Paint" action from the Flutter Inspector in Android
          // Studio, or the "Toggle Debug Paint" command in Visual Studio Code)
          // to see the wireframe for each widget.
          //
          // Column has various properties to control how it sizes itself and
          // how it positions its children. Here we use mainAxisAlignment to
          // center the children vertically; the main axis here is the vertical
          // axis because Columns are vertical (the cross axis would be
          // horizontal).
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Text(
              'You have pushed the button this many times:',
            ),
            Text(
              '$_counter',
              style: Theme.of(context).textTheme.headline4,
            ),
            CurvedCard(
                name: 'Enamul Haque',
                bloodGroup: 'B+ve',
                phone: '+8801843498350',
                email: 'enamul@atilimited.net',
                gender: 'Male',
                id: '505 843 7347',
                birthDate: 'January 12, 1996',
                status: 0,
                statusDetails: 'Pending for approval',
                imageUrl:'https://enamulhaque028.github.io/profile/img/profile.jpg',
            ),
            FlipVisitingCard(
              visitingCardColor: Colors.red,
              assetIcon1: 'assets/images/stethoscope.png',
              assetIcon2: 'assets/images/ambulance.png',
              assetIcon3: 'assets/images/bed.png',
              name: 'Dr. Mourice Tyler',
              designation: 'Heart Specialist',
              phoneNo: '+8801843498350\n+8801843498350',
              email: 'enamul@atilimited.net\nenamul@atilimited.net',
              address: 'ATI Limited\nGareeb-E-Nawaz-Ave',
              institutionShortName: 'DMC',
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: Icon(Icons.add),
      ), // This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}


lib/custom_design.dart
library custom_design;
import 'package:flutter/material.dart';
export 'card_design/view/build_card_curve_line.dart';
export 'visiting_card/visiting_card.dart';

class CustomAlertBox {
  /// show a alert box
  static Future showCustomAlertBox({
    @required BuildContext context,
    @required Widget willDisplayWidget,
  }) {
    assert(context != null, "context is null!!");
    assert(willDisplayWidget != null, "willDisplayWidget is null!!");
    return showDialog(
        context: context,
        builder: (context) {
          return AlertDialog(
            shape: RoundedRectangleBorder(
              borderRadius: BorderRadius.all(Radius.circular(15)),
            ),
            content: Column(
              mainAxisSize: MainAxisSize.min,
              children: <Widget>[
                willDisplayWidget,
                MaterialButton(
                  color: Colors.green,
                  child: Text('close alert', style: TextStyle(color: Colors.white),),
                  onPressed: () {
                    Navigator.of(context).pop();
                  },
                )
              ],
            ),
            elevation: 10,
          );
        });
  }
}

lib/visiting_card/visiting_card.dart
library visiting_card;
import 'package:flutter/material.dart';
import 'back_visiting_card_view.dart';
import 'front_visiting_card_view.dart';

class FlipVisitingCard extends StatefulWidget {
  //front
  ///supports only blue, red and orange [ex. Colors.red]
  final Color visitingCardColor;
  final String institutionLogoUrl;
  final String institutionName;
  final String slogan;
  final String assetIcon1;
  final String assetIcon2;
  final String assetIcon3;
  //back
  final String name;
  final String designation;
  final String backgroundImageUrl;
  final String institutionShortName;
  final String phoneNo;
  final String email;
  final String address;
  const FlipVisitingCard({
    Key key,
    this.visitingCardColor = Colors.red,
    this.institutionLogoUrl = 'https://upload.wikimedia.org/wikipedia/en/a/a1/Dhaka_Medical_College_and_Hospital_logo.png',
    this.institutionName = 'Dhaka Medical College',
    this.slogan = 'YOUR SLOGAN',
    @required this.assetIcon1,
    @required this.assetIcon2,
    @required this.assetIcon3,
    this.backgroundImageUrl = 'https://media.istockphoto.com/photos/medical-stethoscope-isolated-on-white-background-picture-id1159847028?k=20&m=1159847028&s=612x612&w=0&h=cvXAgMmEv_L65fw7jBG6Kqs3GpYXCp8hjoYBBHL4KK8=',
    @required this.name,
    @required this.designation,
    @required this.address,
    @required this.email,
    @required this.phoneNo,
    @required this.institutionShortName,
  }) : super(key: key);
  
  FlipVisitingCardState createState() => new FlipVisitingCardState();
}

class FlipVisitingCardState extends State<FlipVisitingCard> with TickerProviderStateMixin {
  AnimationController _controller;
  Animation<double> _frontScale;
  Animation<double> _backScale;

  @override
  void initState() {
    super.initState();
    _controller = new AnimationController(
      vsync: this,
      duration: const Duration(seconds: 1),
    );
    _frontScale = new Tween(
      begin: 1.0,
      end: 0.0,
    ).animate(new CurvedAnimation(
      parent: _controller,
      curve: new Interval(0.0, 0.5, curve: Curves.easeIn),
    ));
    _backScale = new CurvedAnimation(
      parent: _controller,
      curve: new Interval(0.5, 1.0, curve: Curves.easeOut),
    );
  }

  @override
  void dispose() {
    _controller.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Center(
      child: GestureDetector(
        onTap: () {
          setState(() {
            if (_controller.isCompleted || _controller.velocity > 0)
              _controller.reverse();
            else
              _controller.forward();
          });
        },
        child: new Stack(
          children: <Widget>[
            new AnimatedBuilder(
              child: FrontVisitingCardView(
                blockColor: widget.visitingCardColor,
                institutionLogoUrl: widget.institutionLogoUrl,
                institutionName: widget.institutionName,
                slogan: widget.slogan,
                assetIcon1: widget.assetIcon1,
                assetIcon2: widget.assetIcon2,
                assetIcon3: widget.assetIcon3,
              ),
              animation: _frontScale,
              builder: (BuildContext context, Widget child) {
                final Matrix4 transform = new Matrix4.identity()
                  ..scale(1.0, _frontScale.value, 1.0);
                return new Transform(
                  transform: transform,
                  alignment: FractionalOffset.center,
                  child: child,
                );
              },
            ),
            new AnimatedBuilder(
              child: BackVisitingCardView(
                blockColor: widget.visitingCardColor,
                backgroundImageUrl: widget.backgroundImageUrl,
                institutionLogoUrl: widget.institutionLogoUrl,
                name: widget.name,
                designation: widget.designation,
                institutionShortName: widget.institutionShortName,
                slogan: widget.slogan,
                phoneNo: widget.phoneNo,
                email: widget.email,
                address: widget.address,
              ),
              animation: _backScale,
              builder: (BuildContext context, Widget child) {
                final Matrix4 transform = new Matrix4.identity()
                  ..scale(1.0, _backScale.value, 1.0);
                return new Transform(
                  transform: transform,
                  alignment: FractionalOffset.center,
                  child: child,
                );
              },
            ),
          ],
        ),
      ),
    );
  }
}


lib/visiting_card/back_visiting_card_view.dart
import 'package:flutter/material.dart';
import 'block_tile_back.dart';

class BackVisitingCardView extends StatelessWidget {
  final Color blockColor;
  final String name;
  final String designation;
  final String institutionLogoUrl;
  final String backgroundImageUrl;
  final String institutionShortName;
  final String slogan;
  final String phoneNo;
  final String email;
  final String address;
  const BackVisitingCardView({
    Key key,
    @required this.blockColor,
    @required this.name,
    @required this.designation,
    @required this.institutionLogoUrl,
    @required this.backgroundImageUrl,
    @required this.institutionShortName,
    @required this.slogan,
    @required this.phoneNo,
    @required this.email,
    @required this.address,
  }) : super(key: key);
  @override
  Widget build(BuildContext context) {
    // var topPadding = MediaQuery.of(context).padding.top;
    var size = MediaQuery.of(context).size;
    Color blockColor1;
    Color blockColor2;
    Color blockColor3;
    if (blockColor == Colors.red) {
      blockColor1 = Colors.red;
      blockColor2 = Colors.red[600];
      blockColor3 = Colors.red[700];
    } else if (blockColor == Colors.orange) {
      blockColor1 = Colors.orange;
      blockColor2 = Colors.orange[600];
      blockColor3 = Colors.orange[700];
    } else if (blockColor == Colors.blue) {
      blockColor1 = Colors.blue;
      blockColor2 = Colors.blue[600];
      blockColor3 = Colors.blue[700];
    }
    return Container(
      height: 300,
      width: size.width,
      padding: EdgeInsets.only(top: 32, left: 8, right: 8),
      child: Card(
        shape: RoundedRectangleBorder(
          borderRadius: BorderRadius.all(
            Radius.circular(15),
          ),
        ),
        child: Column(
          children: [
            Stack(
              children: [
                ClipRRect(
                  borderRadius: BorderRadius.only(
                    topLeft: Radius.circular(15),
                    topRight: Radius.circular(15),
                  ),
                  child: Container(
                    height: 150,
                    width: size.width,
                    decoration: BoxDecoration(
                      image: DecorationImage(
                        fit: BoxFit.cover,
                        image: NetworkImage(
                            backgroundImageUrl),
                      ),
                    ),
                  ),
                ),
                Container(
                  height: 150,
                  width: size.width,
                  decoration: BoxDecoration(
                    borderRadius: BorderRadius.only(
                      topLeft: Radius.circular(15),
                      topRight: Radius.circular(15),
                    ),
                    color: Colors.black38,
                  ),
                  child: Padding(
                    padding: const EdgeInsets.fromLTRB(16, 16, 16, 0),
                    child: Row(
                      mainAxisAlignment: MainAxisAlignment.spaceBetween,
                      children: [
                        Column(
                          crossAxisAlignment: CrossAxisAlignment.start,
                          children: [
                            Text(
                              name,
                              style: TextStyle(
                                fontSize: 18,
                                color: Colors.white,
                                fontWeight: FontWeight.bold,
                              ),
                            ),
                            Text(
                              designation,
                              style: TextStyle(
                                // fontSize: 18,
                                color: Colors.white,
                                fontWeight: FontWeight.w400,
                              ),
                            ),
                          ],
                        ),
                        Row(
                          crossAxisAlignment: CrossAxisAlignment.start,
                          children: [
                            Image.network(
                              institutionLogoUrl,
                              height: 60,
                              width: 60,
                            ),
                            Padding(
                              padding: const EdgeInsets.only(top: 8.0),
                              child: Column(
                                crossAxisAlignment: CrossAxisAlignment.start,
                                children: [
                                  Text(
                                    institutionShortName,
                                    style: TextStyle(
                                      fontSize: 18,
                                      color: Colors.white,
                                      fontWeight: FontWeight.bold,
                                    ),
                                  ),
                                  Text(
                                    slogan,
                                    style: TextStyle(
                                      // fontSize: 18,
                                      color: Colors.white,
                                      fontWeight: FontWeight.w400,
                                    ),
                                  ),
                                ],
                              ),
                            ),
                          ],
                        ),
                      ],
                    ),
                  ),
                ),
              ],
            ),
            Container(
              height: 110,
              width: size.width,
              child: Row(
                // mainAxisAlignment: MainAxisAlignment.center,
                children: [
                  BlockTileBack(
                    // color: blockColor,
                    color: blockColor1,
                    bottomLeftRadius: 15,
                    iconData: Icons.phone_android,
                    text: phoneNo,
                  ),
                  BlockTileBack(
                    // color: blockColor.withRed(240).withGreen(30).withBlue(20),
                    color: blockColor2,
                    iconData: Icons.email_outlined,
                    text: email,
                  ),
                  BlockTileBack(
                    // color: blockColor.withRed(200).withGreen(30).withBlue(20),
                    color: blockColor3,
                    bottomRightRadius: 15,
                    iconData: Icons.location_on_outlined,
                    text: address,
                  ),
                ],
              ),
            ),
          ],
        ),
      ),
    );
  }
}


lib/visiting_card/front_visiting_card_view.dart
import 'package:flutter/material.dart';

import 'block_tile_front.dart';

class FrontVisitingCardView extends StatelessWidget {
  final Color blockColor;
  final String institutionLogoUrl;
  final String institutionName;
  final String slogan;
  final String assetIcon1;
  final String assetIcon2;
  final String assetIcon3;
  const FrontVisitingCardView({
    Key key,
    @required this.blockColor,
    @required this.institutionLogoUrl,
    @required this.institutionName,
    @required this.slogan,
    @required this.assetIcon1,
    @required this.assetIcon2,
    @required this.assetIcon3,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    // var topPadding = MediaQuery.of(context).padding.top;
    var size = MediaQuery.of(context).size;
    Color blockColor1;
    Color blockColor2;
    Color blockColor3;
    if (blockColor == Colors.red) {
      blockColor1 = Colors.red;
      blockColor2 = Colors.red[600];
      blockColor3 = Colors.red[700];
    } else if (blockColor == Colors.orange) {
      blockColor1 = Colors.orange;
      blockColor2 = Colors.orange[600];
      blockColor3 = Colors.orange[700];
    } else if (blockColor == Colors.blue) {
      blockColor1 = Colors.blue;
      blockColor2 = Colors.blue[600];
      blockColor3 = Colors.blue[700];
    }
    return Container(
      height: 300,
      width: size.width,
      padding: EdgeInsets.only(top: 32, left: 8, right: 8),
      child: Card(
        shape: RoundedRectangleBorder(
          borderRadius: BorderRadius.all(
            Radius.circular(15),
          ),
        ),
        child: Row(
          children: [
            Container(
              height: 300,
              width: size.width / 2,
              decoration: BoxDecoration(
                borderRadius: BorderRadius.only(
                  topLeft: Radius.circular(15),
                  bottomLeft: Radius.circular(15),
                ),
                color: Color(0xFF29303a),
              ),
              child: Padding(
                padding: const EdgeInsets.symmetric(horizontal: 8.0),
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.center,
                  mainAxisAlignment: MainAxisAlignment.center,
                  children: [
                    Image.network(
                      institutionLogoUrl,
                      height: 60,
                      width: 60,
                    ),
                    Text(
                      institutionName,
                      style: TextStyle(
                        fontSize: 16,
                        color: Colors.white,
                        fontWeight: FontWeight.bold,
                      ),
                    ),
                    Text(
                      slogan,
                      style: TextStyle(
                        // fontSize: 18,
                        color: Colors.white,
                        fontWeight: FontWeight.w400,
                      ),
                    ),
                  ],
                ),
              ),
            ),
            Column(
              children: [
                BlockTileFront(
                  color: blockColor3,
                  assetIcon: assetIcon1,
                  text: 'GET HEALTH\nRESULTS',
                  topRightRadius: 15,
                ),
                BlockTileFront(
                  color: blockColor2,
                  assetIcon: assetIcon2,
                  text: 'URGENCY\nSERVICES',
                ),
                BlockTileFront(
                  color: blockColor1,
                  assetIcon: assetIcon3,
                  text: 'PRIVET\nRESULTS',
                  bottomRightRadius: 15,
                  height: 88,
                ),
              ],
            ),
          ],
        ),
      ),
    );
  }
}


lib/card_design/paint/custom_paint_curved_card.dart
import 'package:flutter/material.dart';
import 'dart:ui' as ui;

class RPSCustomPainter extends CustomPainter {
  @override
  void paint(Canvas canvas, Size size) {
    Paint paint_0 = new Paint()
      // ..color = Color.fromARGB(205, 47, 47, 225)
      ..style = PaintingStyle.fill
      ..strokeWidth = 1;
    paint_0.shader = ui.Gradient.linear(
      Offset(size.width * 0.64, size.height * 0.80),
      Offset(0, size.height * 1.00),
      [Colors.white, Colors.white.withOpacity(0.2)],
      [1.0, 1.0],
      //TileMode.repeated,
    );

    Path path_0 = Path();
    path_0.moveTo(0, 0);
    path_0.lineTo(0, size.height * 0.1008500);
    path_0.lineTo(size.width * 0.1400000, size.height * 0.1107000);
    path_0.lineTo(size.width * 0.3585200, size.height * 0.6497500);
    path_0.lineTo(size.width, size.height * 0.6532000);
    path_0.lineTo(size.width, 0);
    path_0.quadraticBezierTo(size.width * 0.7485000, 0, 0, 0);
    path_0.close();

    canvas.drawPath(path_0, paint_0);

    Paint paint_1 = new Paint()
      // ..color = Color.fromARGB(216, 209, 209, 214)
      ..style = PaintingStyle.fill
      ..strokeWidth = 1;

    paint_1.shader = ui.Gradient.linear(
      Offset(size.width * 0.64, size.height * 0.80),
      Offset(0, size.height * 1.00),
      [Colors.grey[300]],
      // [Colors.black12],
      [1.0],
      //TileMode.repeated,
    );

    Path path_1 = Path();
    path_1.moveTo(size.width * 0.3571000, size.height * 0.6465000);
    path_1.lineTo(size.width * 0.4380000, size.height * 0.8462500);
    path_1.lineTo(size.width, size.height * 0.8498000);
    path_1.lineTo(size.width, size.height * 0.6498500);
    path_1.lineTo(size.width * 0.3571000, size.height * 0.6465000);
    path_1.close();

    canvas.drawPath(path_1, paint_1);

    Paint paint_2 = new Paint()
      // ..color = Color.fromARGB(157, 163, 19, 19)
      ..style = PaintingStyle.fill
      ..strokeWidth = 1;

    paint_2.shader = ui.Gradient.linear(
      Offset(size.width * 0.64, size.height * 0.80),
      Offset(0, size.height * 1.00),
      [Colors.grey[200]],
      [1.0],
      //TileMode.repeated,
    );

    Path path_2 = Path();
    path_2.moveTo(0, size.height * 0.1607500);
    path_2.lineTo(size.width * 0.1238400, size.height * 0.1642000);
    path_2.lineTo(size.width * 0.4238200, size.height * 0.9036500);
    path_2.lineTo(size.width, size.height * 0.9071000);
    path_2.lineTo(size.width, size.height * 0.8486000);
    path_2.lineTo(size.width * 0.4380000, size.height * 0.8450000);
    path_2.lineTo(size.width * 0.1379800, size.height * 0.1036500);
    path_2.lineTo(0, size.height * 0.1000000);
    path_2.lineTo(0, size.height * 0.1607500);
    path_2.close();

    canvas.drawPath(path_2, paint_2);
  }

  @override
  bool shouldRepaint(covariant CustomPainter oldDelegate) {
    return true;
  }
}

lib/card_design/view/build_card_curve_line.dart
library build_card_curve_line;
import 'dart:math';
import 'package:custom_design/card_design/paint/custom_paint_curved_card.dart';
import 'package:custom_design/card_design/utils.dart';
import 'package:flutter/material.dart';

class CurvedCard extends StatelessWidget {
  final String name;
  final String bloodGroup;
  final String phone;
  final String email;
  final String gender;
  final String id;
  final String birthDate;
  final int status;
  final String statusDetails;
  final String imageUrl;

  const CurvedCard({
    Key key,
    @required this.name,
    @required this.bloodGroup,
    @required this.phone,
    @required this.email,
    @required this.gender,
    @required this.id,
    @required this.birthDate,
    this.status = 0,
    @required this.statusDetails,
    @required this.imageUrl,
  }) : super(key: key);


  @override
  Widget build(BuildContext context) {
    var width = MediaQuery.of(context).size.width;
    return Container(
      padding: EdgeInsets.only(top: 10, left: 5.0, right: 5.0),
      height: 230,
      child: Stack(
        children: [
          Card(
            elevation: 4.0,
            shape: RoundedRectangleBorder(
              borderRadius: BorderRadius.circular(15.0),
            ),
            child: Container(
              width: width,
              decoration: BoxDecoration(
                // color: cSky,
                borderRadius: BorderRadius.circular(10.0),
                gradient: LinearGradient(
                  colors: <Color>[
                    Colors.primaries[Random().nextInt(Colors.primaries.length)],
                    // Colors.primaries[Random().nextInt(Colors.primaries.length)]
                    //     .withOpacity(0.1),
                    Colors.white38,
                    Colors.white
                  ],
                ),
              ),
              child: CustomPaint(
                size: Size(
                    500,
                    (500 * 0.4)
                        .toDouble()), //You can Replace [WIDTH] with your desired width for Custom Paint and height will be calculated automatically
                painter: RPSCustomPainter(),
              ),
            ),
          ),
          Positioned(
            top: 55,
            left: 10,
            child: Padding(
              padding: const EdgeInsets.all(8.0),
              child: CircleAvatar(
                backgroundColor: Colors.white,
                radius: 30,
                backgroundImage: NetworkImage(imageUrl),
              ),
            ),
          ),
          Positioned(
            left: width * 0.32,
            child: Container(
              width: width * 0.65,
              child: Row(
                mainAxisAlignment: MainAxisAlignment.spaceBetween,
                children: [
                  Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      SizedBox(height: 22),
                      Text(
                        name.toUpperCase(),
                        style: TextStyle(
                            fontSize: 18, fontWeight: FontWeight.bold),
                      ),
                      SizedBox(height: 5),
                      Row(
                        children: [
                          Text(
                            birthDate,
                            style: TextStyle(
                                color: Colors.grey,
                                fontWeight: FontWeight.w500,
                            ),
                          ),
                          Container(
                            height: 20,
                            padding: const EdgeInsets.only(left: 8, right: 8),
                            child: VerticalDivider(
                                color: Colors.grey, thickness: 1.2),
                          ),
                          Text(
                            gender,
                            style: TextStyle(
                                color: Colors.grey,
                                fontWeight: FontWeight.w500),
                          ),
                          Container(
                            height: 20,
                            padding: const EdgeInsets.only(left: 8, right: 8),
                            child: VerticalDivider(
                                color: Colors.grey, thickness: 1.2),
                          ),
                          Text(
                            bloodGroup.toUpperCase(),
                            style: TextStyle(
                                color: Colors.grey,
                                fontWeight: FontWeight.w500),
                            textAlign: TextAlign.center,
                          ),
                        ],
                      ),
                      SizedBox(height: 4),
                      Row(
                        // mainAxisAlignment: MainAxisAlignment.spaceBetween,
                        children: [
                          Icon(
                            Icons.phone,
                            color: UiColor.cViolet,
                          ),
                          SizedBox(width: 4),
                          Text(
                            phone,
                            overflow: TextOverflow.ellipsis,
                            maxLines: 1,
                            style: TextStyle(
                              fontSize: 15,
                              fontWeight: FontWeight.w500,
                              color: Colors.grey,
                            ),
                          ),
                        ],
                      ),
                      // SizedBox(height: 8),
                      Row(
                        // mainAxisAlignment: MainAxisAlignment.spaceBetween,
                        children: [
                          Icon(
                            Icons.email,
                            color: UiColor.cViolet,
                          ),
                          SizedBox(width: 4),
                          Container(
                            width: 200,
                            child: Text(
                              email.toLowerCase(),
                              overflow: TextOverflow.ellipsis,
                              maxLines: 1,
                              style: TextStyle(
                                fontSize: 15,
                                fontWeight: FontWeight.w500,
                                color: Colors.grey,
                              ),
                            ),
                          ),
                        ],
                      ),
                    ],
                  ),
                  status == 0
                      ? Center(
                          child: RotatedBox(
                            quarterTurns: 1,
                            child: Text(
                              'Requested',
                              style: TextStyle(color: Colors.red),
                            ),
                          ),
                        )
                      : Container(),
                ],
              ),
            ),
          ),
          Positioned(
            left: width * 0.38,
            top: 131,
            child: Padding(
              padding:
                  EdgeInsets.only(top: 10.0, bottom: 5.0, left: 20, right: 5),
              child: Container(
                // padding: EdgeInsets.only(left: 5, right: 5),
                width: width * 0.6,
                decoration: new BoxDecoration(
                  border: Border.symmetric(
                    horizontal: BorderSide(
                      width: 2,
                      color: Colors.transparent,
                    ),
                  ),
                ),
                child: Text(
                  statusDetails,
                  style: TextStyle(
                    fontSize: 18,
                    fontWeight: FontWeight.bold,
                    color: status == 0 ? Colors.red : Colors.green,
                  ),
                ),
              ),
            ),
          ),
        ],
      ),
    );
  }
}
